src/pages/index.js

```diff
@@ -41,38 +59,77 @@ function handleImageClick(name, link) {
 }
 
 //Popup Forms
-const profilePopup = new PopupWithForm(profileModal, (data) => {
-  console.log(data);
+const profilePopup = new PopupWithForm(profileModal, (data, button) => {
   userInfo.setUserInfo(data);
-  profilePopup.closePopup();
-});
-const cardForm = new PopupWithForm(newCardModal, (data) => {
-  const name = newCardTitle.value;
-  const link = newCardURL.value;
-  addCard({ name, link }, mediaList);
-  cardForm.closePopup();
+  api.saveUserInfo(data).then(() => {
+    button.textContent = "Save";
```

Comment:
You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case: `success` or `failure`.


**Please, correct it everywhere**